### PR TITLE
Feat/verify receive address

### LINKED ISSUE #9775: Device receive - read address from device and confirm it
## Description
If a user has a connected authorized device and navigates to the receive screen, he can see the "device screen" mockup containing the unverified address. If he clicks on the button `view address`, then the connected device shows the address and asks the user for verification. If is the verification successful, the address QR code with a success message is displayed.

This issue should make the foundational changes so the other task of the [[EPIC] Receive flow with connected Trezor Device](https://github.com/trezor/trezor-suite/issues/9186#top) can be implemented. 

> Caution! This issue does not implement other things than those described in the acceptance criteria section. Issues like informational bottom sheets or images will be resolved in follow-up issues from the same EPIC.

 

**Design:** [Figma](https://www.figma.com/file/IidkuNfa5OWtPgta9jF8Gl/T-%E2%80%A2-Handoff?type=design&node-id=3187-1483&mode=design&t=6L0X5nhfiHuuThI4-4)

## Acceptance criteria:
- if a user has connected the device and navigates to receive flow, he see the device screen mock with a unverified address in it
- if the device connection is not supported (iOS), the receive flow is unchanged 
-  a user can view the address on the connected device and confirm it
- when is the device confirmation successful the address QR code is displayed

### PR DESCRIPTION
**Pull request by PeKne**
## Description

First PR related to the receive EPIC that resolves requirements of this specific issue #9775. There will be other functionalities implemented as part of the other issues from recieve epic, so expect that there are still things that will change soon.

Also many changes happened, so I recommend doing the review commit by commit. 

## Related Issue

Resolve #9775

## Screenshots:

https://github.com/trezor/trezor-suite/assets/26143964/c32be6de-3c26-4625-8809-d99d4c3dfbd8

### CODE REVIEW

File: suite-native/receive/assets/confirmOnTrezor.png
Comment (vytick) [inline suite-native/receive/assets/confirmOnTrezor.png:1]: I suggest to remove it from the image and make it in code, because the text in the image is nontranslatable.

File: suite-native/receive/src/components/ReceiveAccount.tsx [hunk 1]
```diff
@@ -1,117 +1,52 @@
...
 
-    const handleGoBack = () => {
-        navigation.goBack();
-    };
-
-    const handleOpenEduLink = () => {
-        openLink('https://trezor.io/learn/a/verifying-trezor-suite-lite-addresses');
-    };
+    if (G.isNullable(account) || G.isNullable(address))
+        return <ErrorMessage errorMessage="Something went wrong" />;
```
Comment (vytick) [inline suite-native/receive/src/components/ReceiveAccount.tsx]: Please use intl

File: suite-native/receive/src/components/ReceiveAccountDetailsCard.tsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+export const ReceiveAccountDetailsCard = ({
+    accountKey,
+    tokenContract,
+}: ReceiveAccountDetailsCardProps) => {
+    const account = useSelector((state: AccountsRootState) =>
+        selectAccountByKey(state, accountKey),
+    );
+
+    if (G.isNullable(account))
+        return <ErrorMessage errorMessage={`Account ${accountKey} not found.`} />;
```
Comment (vytick) [inline suite-native/receive/src/components/ReceiveAccountDetailsCard.tsx]: intl please

File: suite-native/receive/src/components/UnverifiedAddressDevice.tsx [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+export const UnverifiedAddressDevice = ({
+    address,
+    isAddressRevealed,
+}: DeviceScreenAddressProps) => {
+    const { applyStyle } = useNativeStyles();
+
+    const addressChunkedLines = pipe(address, splitAddressToChunks, A.splitEvery(4));
+
+    const deviceFont = useFont(
+        require('../../../../packages/theme/fonts/RobotoMono-Regular.ttf'),
```
Comment (vytick) [inline suite-native/receive/src/components/UnverifiedAddressDevice.tsx:69]: I think katka has said on standup, that the fonts differ for models. Maybe to make this into constant similar to FONT_SIZE and handle later?

File: suite-native/receive/src/screens/ReceiveModalScreen.tsx [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+
+const ReceiveModalScreenSubHeader = ({ accountKey, tokenContract }: ScreenSubHeaderContent) => {
+    const accountLabel = useSelector((state: AccountsRootState) =>
+        selectAccountLabel(state, accountKey),
+    );
+    const networkSymbol = useSelector((state: AccountsRootState) =>
+        selectAccountNetworkSymbol(state, accountKey),
+    );
+    const tokenSymbol = useSelector((state: AccountsRootState) =>
+        selectEthereumAccountTokenSymbol(state, accountKey, tokenContract),
```
Comment (vytick) [inline suite-native/receive/src/screens/ReceiveModalScreen.tsx:40]: i suggest to use `ethereumTokenSymbol` because there probably will be other tokens once Solana is merged

**Review by vytick — CHANGES_REQUESTED** [review]
Nice work :) just a few small comments:

File: suite-common/wallet-utils/src/confirmAddressOnDevice.ts [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+
+import {
+    getAddressType,
+    getStakingPath,
+    getProtocolMagic,
+    getNetworkId,
+    getDerivationType,
+} from './cardanoUtils';
+
+export const confirmAddressOnDevice = async ({
```
Comment (juriczech) [inline suite-common/wallet-utils/src/confirmAddressOnDevice.ts]: I would make this a thunk with access to state and all of these params can be removed as they can be accessed through selector (apart from address path I guess?)

File: suite-common/wallet-utils/src/confirmAddressOnDevice.ts [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+    device,
+    account,
+    addressPath,
+}: {
+    device: TrezorDevice;
+    account: Account;
+    addressPath: string;
+}) => {
+    let response;
```
Comment (juriczech) [inline suite-common/wallet-utils/src/confirmAddressOnDevice.ts]: nit: no need for this? You can just straight away return the response from each condition

File: suite-common/wallet-utils/src/cardanoUtils.ts [hunk 7]
```diff
@@ -15,3 +17,15 @@ export const getDerivationType = (accountType: Network['accountType']) => {
...
     }
 };
+
+export const getStakingPath = (account: Account) => `m/1852'/1815'/${account.index}'/2/0`;
+
+export const getProtocolMagic = (accountSymbol: Account['symbol']) =>
+    // TODO: use testnet magic from connect once this PR is merged https://github.com/trezor/connect/pull/1046
+    accountSymbol === 'ada' ? CARDANO.PROTOCOL_MAGICS.mainnet : 1097911063;
+
+export const getAddressType = (_accountType: Account['accountType']) =>
```
Comment (juriczech) [inline suite-common/wallet-utils/src/cardanoUtils.ts]: Please remove this unused param

File: suite-native/app/src/navigation/AppTabNavigator.tsx [hunk 8]
```diff
@@ -24,7 +24,7 @@ export const AppTabNavigator = () => (
         >
             <Tab.Screen name={AppTabsRoutes.HomeStack} component={HomeStackNavigator} />
             <Tab.Screen name={AppTabsRoutes.AccountsStack} component={AccountsStackNavigator} />
-            <Tab.Screen name={AppTabsRoutes.ReceiveStack} component={ReceiveStackNavigator} />
+            <Tab.Screen name={AppTabsRoutes.ReceiveAccounts} component={ReceiveAccountsScreen} />
```
Comment (juriczech) [inline suite-native/app/src/navigation/AppTabNavigator.tsx]: Why this change? I would keep this to follow the convention to export a navigator from a module that's part of the AppTabsNavigator. Same thing as HomeStackNavigator

File: suite-native/receive/src/components/ConfirmOnTrezorImage.tsx [hunk 9]
```diff
@@ -0,0 +1,37 @@
+import { Image, Pressable } from 'react-native';
+import Animated, { FadeIn } from 'react-native-reanimated';
+import React from 'react';
+
+import { prepareNativeStyle, useNativeStyles } from '@trezor/styles';
+import { useToast } from '@suite-native/toasts';
+
+const imageContainerStyle = prepareNativeStyle(utils => ({
+    position: 'absolute',
+    bottom: -utils.spacings.medium, // Hides a part of the image under bottom screen edge.
```
Comment (juriczech) [inline suite-native/receive/src/components/ConfirmOnTrezorImage.tsx:10]: Why is this necessary? Isn't it just a hack to avoid Screen.tsx spacings and paddings?

File: suite-common/wallet-core/src/accounts/accountsReducer.ts [hunk 10]
```diff
@@ -137,7 +137,14 @@ export const selectMainnetAccounts = memoize((state: AccountsRootState) =>
     ),
 );
 
-export const selectAccountByKey = (state: AccountsRootState, accountKey: AccountKey) => {
+export const selectNumberOfAccounts = (state: AccountsRootState) => selectAccounts(state).length;
+
+export const selectUserHasAccounts = (state: AccountsRootState): boolean =>
+    pipe(selectAccounts(state), A.isNotEmpty);
+
+export const selectAccountByKey = (state: AccountsRootState, accountKey?: AccountKey) => {
```
Comment (Nodonisko) [inline suite-common/wallet-core/src/accounts/accountsReducer.ts:155]: Why did you made key optional?

File: suite-native/receive/src/components/ReceiveAccount.tsx [hunk 11]
```diff
@@ -1,117 +1,52 @@
...
-            {tokenContract && (
-                <AlertBox
-                    variant="info"
-                    title="Your receive address is your Ethereum address."
-                    isStandalone
+                <ReceiveAddressCard
+                    address={address}
+                    isReceiveApproved={isReceiveApproved}
+                    isUnverifiedAddressRevealed={isUnverifiedAddressRevealed}
+                    onShowAddress={handleShowAddress}
```
Comment (juriczech) [inline suite-native/receive/src/components/ReceiveAccount.tsx:47]: Pass in the accountKey and select these values from `useAccountReceiveAddress` please.

File: suite-native/app/src/navigation/routes.ts [hunk 12]
```diff
@@ -17,8 +17,8 @@ const accountsStack = enhanceTabOption({
     },
 });
 
-const receiveStack = enhanceTabOption({
-    routeName: AppTabsRoutes.ReceiveStack,
+const receiveAccounts = enhanceTabOption({
```
Comment (Nodonisko) [inline suite-native/app/src/navigation/routes.ts]: I would probably keep suffix `Stack`

File: suite-native/helpers/src/splitAddressToChunks.ts [hunk 13]
```diff
@@ -0,0 +1,4 @@
+import { pipe, S, A } from '@mobily/ts-belt';
+
+export const splitAddressToChunks = (address: string): readonly string[] =>
```
Comment (Nodonisko) [inline suite-native/helpers/src/splitAddressToChunks.ts]: would be nice to have tests for this one

File: suite-native/receive/src/components/UnverifiedAddressWarning.tsx [hunk 14]
```diff
@@ -0,0 +1,28 @@
+import { Box, VStack, Text, Pictogram, TrezorSuiteLiteHeader } from '@suite-native/atoms';
+import { useTranslate, Translation } from '@suite-native/intl';
+
+export const UnverifiedAddressWarning = () => {
+    const { translate } = useTranslate();
+    return (
+        <VStack spacing="medium" paddingVertical="extraLarge">
```
Comment (juriczech) [inline suite-native/receive/src/components/UnverifiedAddressWarning.tsx]: Redundant VStack

File: suite-native/receive/src/components/ReceiveAddressCard.tsx [hunk 15]
```diff
@@ -0,0 +1,38 @@
...
+    address,
+    isUnverifiedAddressRevealed,
+    isReceiveApproved,
+    onShowAddress,
+}: ReceiveAddressCardProps) => (
+    <Animated.View layout={Layout}>
+        {/* TODO: display card alert according to design */}
+        {/* https://github.com/trezor/trezor-suite/issues/9776 */}
+        <Card>
+            <VStack spacing="medium" paddingVertical="small">
```
Comment (juriczech) [inline suite-native/receive/src/components/ReceiveAddressCard.tsx]: Place the VStack inside UnverifiedAddress as it doesn't affect AddressQRCode please.

File: suite-native/receive/src/components/UnverifiedAddressDevice.tsx [hunk 16]
```diff
@@ -0,0 +1,104 @@
...
+    height: DEVICE_SCREEN_HEIGHT,
+}));
+
+const AddressLineGradient = ({ isApplied }: { isApplied: boolean }) => (
+    <LinearGradient
+        start={vec(FADEOUT_GRADIENT_CENTER, 0)}
+        end={vec(FADEOUT_GRADIENT_CENTER, FADEOUT_GRADIENT_HEIGHT)}
+        colors={[DEVICE_TEXT_COLOR, isApplied ? DEVICE_SCREEN_BACKGROUND_COLOR : DEVICE_TEXT_COLOR]}
+    />
+);
```
Comment (juriczech) [inline suite-native/receive/src/components/UnverifiedAddressDevice.tsx:58]: nit: Seems redundant to me as it's not reused and doesn't save any code

File: suite-native/receive/src/components/ReceiveAccount.tsx (hunk 11 above)
Comment (juriczech) [reply suite-native/receive/src/components/ReceiveAccount.tsx:47]: Same rule for all children components. I guess that it will be better to use atom rather than state to move that from the hook

File: suite-native/receive/src/components/UnverifiedAddressDevice.tsx [hunk 17]
```diff
@@ -0,0 +1,104 @@
...
+                                    x={0}
+                                    y={(index + 1) * LINE_HEIGHT}
+                                    text={lineString}
+                                    font={deviceFont}
+                                    color={DEVICE_TEXT_COLOR}
+                                >
+                                    <AddressLineGradient isApplied={isLineFaded} />
+                                </SkiaText>
+                            );
+                        })}
```
Comment (juriczech) [inline suite-native/receive/src/components/UnverifiedAddressDevice.tsx:97]: Why not create a component for this? Makes more sense than AddressLineGradient, which then you can just paste inside IMO

File: suite-native/receive/src/components/ReceiveAccount.tsx (hunk 11 above)
Comment (juriczech) [reply suite-native/receive/src/components/ReceiveAccount.tsx:47]: Or maybe use context? However there seems to be quite a lot of props passing around

File: suite-native/receive/src/screens/ReceiveModalScreen.tsx [hunk 18]
```diff
@@ -0,0 +1,97 @@
...
+import { selectEthereumAccountTokenSymbol } from '@suite-native/ethereum-tokens';
+
+import { ReceiveAccount } from '../components/ReceiveAccount';
+
+type ScreenSubHeaderContent = {
+    accountKey?: AccountKey;
+    tokenContract?: TokenAddress;
+};
+
+const ReceiveModalScreenSubHeader = ({ accountKey, tokenContract }: ScreenSubHeaderContent) => {
```
Comment (juriczech) [inline suite-native/receive/src/screens/ReceiveModalScreen.tsx:32]: Why do you have it in this file?

File: suite-native/receive/src/components/UnverifiedAddressDevice.tsx [hunk 19]
```diff
@@ -0,0 +1,104 @@
...
+
+export const UnverifiedAddressDevice = ({
+    address,
+    isAddressRevealed,
+}: DeviceScreenAddressProps) => {
+    const { applyStyle } = useNativeStyles();
+
+    const addressChunkedLines = pipe(address, splitAddressToChunks, A.splitEvery(4));
+
+    const deviceFont = useFont(
```
Comment (Nodonisko) [inline suite-native/receive/src/components/UnverifiedAddressDevice.tsx:68]: I am not sure if we should do this here, maybe place it on top level as we did for other fonts?

File: suite-native/receive/src/screens/ReceiveModalScreen.tsx [hunk 20]
```diff
@@ -0,0 +1,97 @@
...
+    ) => {
+        navigation.dispatch({
+            ...CommonActions.setParams({
+                accountKey: selectedAccountKey,
+                tokenContract: selectedTokenContract,
+            }),
+        });
+    };
+
+    const isAccountSelected = G.isNotNullable(accountKey);
```
Comment (juriczech) [inline suite-native/receive/src/screens/ReceiveModalScreen.tsx:79]: Do you need to handle this through navigation params? Seems really weird. Is this just hack to reuse the goBack functionality?

Comment (juriczech) [reply suite-native/receive/src/screens/ReceiveModalScreen.tsx:79]: Why not add ReceiveAccount to RootStack? That way you will solve most of these navigation problems and still avoid using TabBar.tsx

**Review by juriczech — CHANGES_REQUESTED** [review]

File: suite-native/qr-code/src/components/AddressQRCode.tsx [hunk 21]
```diff
@@ -49,17 +48,17 @@ export const AddressQRCode = ({ address, backgroundElevation = '0' }: AddressQRC
                     size="small"
                     iconLeft="copy"
                     onPress={handleCopyAddress}
-                    colorScheme={`tertiaryElevation${backgroundElevation}`}
+                    colorScheme="tertiaryElevation1"
                 >
-                    Copy
+                    {translate('moduleReceive.receiveAddressCard.qrCode.copy.button')}
```
Comment (Nodonisko) [inline suite-native/qr-code/src/components/AddressQRCode.tsx]: This strings are not part of `moduleReceive` but they are universal to package `qr-code`

File: suite-native/navigation/src/routes.ts [hunk 22]
```diff
@@ -13,7 +13,7 @@ export enum RootStackRoutes {
 export enum AppTabsRoutes {
     HomeStack = 'HomeStack',
     AccountsStack = 'AccountsStack',
-    ReceiveStack = 'ReceiveStack',
+    ReceiveAccounts = 'ReceiveAccounts',
```
Comment (Nodonisko) [inline suite-native/navigation/src/routes.ts]: Why not `Stack`?

**Review by Nodonisko — CHANGES_REQUESTED** [review]

**Comment by AdamSchinzel** [comment]
Not sure if it's on purpose but I see that the chunkify feature is not complete (missing pagination, fonts, settings etc). Feel free to reach out when you will be doing it because I was implementing this to Suite.

File: suite-common/wallet-utils/src/cardanoUtils.ts (hunk 7 above)
Comment (PeKne) [reply suite-common/wallet-utils/src/cardanoUtils.ts]: Fixed by [b7a1390](https://github.com/trezor/trezor-suite/pull/9844/commits/b7a13905102b323b774316dcdd4587fdbe8ae615). I had to move whole `cardanoUtils` to the suite-common, to be able fix its tests. But we will need all of it soon anyway.

File: suite-native/receive/assets/confirmOnTrezor.png
Comment (PeKne) [reply suite-native/receive/assets/confirmOnTrezor.png:1]: Sure, you are right. This will be revisited in a follow up issue. The image is just a placeholder for now.

File: suite-native/receive/src/components/UnverifiedAddressDevice.tsx (hunk 3 above)
Comment (PeKne) [reply suite-native/receive/src/components/UnverifiedAddressDevice.tsx:69]: Yes, please ignore it for now. The structure of this file will be revisited with the follow up issue.

File: suite-native/receive/src/screens/ReceiveModalScreen.tsx (hunk 4 above)
Comment (PeKne) [reply suite-native/receive/src/screens/ReceiveModalScreen.tsx:40]: fixed in 8af235d6e7

File: suite-native/receive/src/components/UnverifiedAddressWarning.tsx (hunk 14 above)
Comment (PeKne) [reply suite-native/receive/src/components/UnverifiedAddressWarning.tsx]: fixed in 8af235d6e7

File: suite-native/receive/src/components/ReceiveAddressCard.tsx (hunk 15 above)
Comment (PeKne) [reply suite-native/receive/src/components/ReceiveAddressCard.tsx]: fixed in 8af235d6e7

File: suite-native/receive/src/components/UnverifiedAddressDevice.tsx (hunk 16 above)
Comment (PeKne) [reply suite-native/receive/src/components/UnverifiedAddressDevice.tsx:58]: This file will be probably redesigned according to final design when ready. Please ignore it for now.

File: suite-native/receive/src/components/UnverifiedAddressDevice.tsx (hunk 17 above)
Comment (PeKne) [reply suite-native/receive/src/components/UnverifiedAddressDevice.tsx:97]: This file will be probably redesigned according to final design when ready. Please ignore it for now.

File: suite-native/receive/src/screens/ReceiveModalScreen.tsx (hunk 18 above)
Comment (PeKne) [reply suite-native/receive/src/screens/ReceiveModalScreen.tsx:32]: Because it should be never used anywhere else than in the `ReceiveModalScreen`. Thus i think it should stay here and never be exported.

File: suite-native/helpers/src/splitAddressToChunks.ts (hunk 13 above)
Comment (PeKne) [reply suite-native/helpers/src/splitAddressToChunks.ts]: fixed in https://github.com/trezor/trezor-suite/commit/8af235d6e7bb7d75e4232baa31ce1ffe754a15b6

File: suite-native/receive/src/components/UnverifiedAddressDevice.tsx (hunk 19 above)
Comment (PeKne) [reply suite-native/receive/src/components/UnverifiedAddressDevice.tsx:68]: You are probably right. Please ignore this, I will refactor in the followp issue that takes care about per-device formatting.

File: suite-native/qr-code/src/components/AddressQRCode.tsx (hunk 21 above)
Comment (PeKne) [reply suite-native/qr-code/src/components/AddressQRCode.tsx]: fixed in https://github.com/trezor/trezor-suite/commit/8af235d6e7bb7d75e4232baa31ce1ffe754a15b6

File: suite-native/receive/src/components/UnverifiedAddressDevice.tsx (hunk 16 above)
Comment (PeKne) [reply suite-native/receive/src/components/UnverifiedAddressDevice.tsx:58]: This file will be probably redesigned according to final design when ready. Please ignore it for now.

File: suite-native/app/src/navigation/AppTabNavigator.tsx (hunk 8 above)
Comment (PeKne) [reply suite-native/app/src/navigation/AppTabNavigator.tsx]: reverted in acc76df2ee

File: suite-native/receive/src/screens/ReceiveModalScreen.tsx (hunk 20 above)
Comment (PeKne) [reply suite-native/receive/src/screens/ReceiveModalScreen.tsx:79]: rootstack reverted in acc76df2ee. This won't be changed now after mutual discussion.

File: suite-native/app/src/navigation/routes.ts (hunk 12 above)
Comment (PeKne) [reply suite-native/app/src/navigation/routes.ts]: reverted in acc76df2ee

File: suite-native/navigation/src/routes.ts (hunk 22 above)
Comment (PeKne) [reply suite-native/navigation/src/routes.ts]: reverted in acc76df2ee

File: suite-common/wallet-core/src/accounts/accountsReducer.ts (hunk 10 above)
Comment (PeKne) [reply suite-common/wallet-core/src/accounts/accountsReducer.ts:155]: Because I'm now using this selector in a components, where the `accountKey` come from a prop as `AccountKey | undefined`. Solution would be to check that the value is present before calling the selector, but selecors are hooks and can not be called conditionally. 

We have possibility to pass undefined to other selectors, I believe this doesn't do any harm.

File: suite-native/receive/src/screens/ReceiveModalScreen.tsx (hunk 18 above)
Comment (juriczech) [reply suite-native/receive/src/screens/ReceiveModalScreen.tsx:32]: The goal is to create reusable components that I don't have to refactor from other files. By moving it to other file:
- it will have a working diff. Now if I need to reuse it, I will have to copy + paste from here and the diff / git blame will be lost
- you will make the visual overhead of this component much better 

The agreement was that it's fine to do it on small scale, but I would say the amount of logic in this would make sense to separate. I don't want to be dogmatic about this but also if I use your argument, we might as well put everything in App.tsx and only create reusable components? Let's draw the line somewhere and follow the convention of 1 file = 1 component 99% of the time and make some exceptions, but no reason here IMO.

File: suite-native/receive/src/components/ReceiveAccount.tsx (hunk 1 above)
Comment (PeKne) [reply suite-native/receive/src/components/ReceiveAccount.tsx]: 99c16e84a5

File: suite-native/receive/src/components/ReceiveAccountDetailsCard.tsx (hunk 2 above)
Comment (PeKne) [reply suite-native/receive/src/components/ReceiveAccountDetailsCard.tsx]: 99c16e84a5

File: suite-native/receive/src/components/ConfirmOnTrezorImage.tsx (hunk 9 above)
Comment (PeKne) [reply suite-native/receive/src/components/ConfirmOnTrezorImage.tsx:10]: This is needed to hide part of the image under the bottom edge of the screen (as the comment says). But I think this image will be removed later anyway. The design is still not ready.

File: suite-native/receive/src/screens/ReceiveModalScreen.tsx (hunk 18 above)
Comment (juriczech) [reply suite-native/receive/src/screens/ReceiveModalScreen.tsx:32]: I'm confused, so let's clarify. I thought we agreed that `ReceiveAccount` component will be separate screen in RootStack for now before we refactor navigation, no? In that case we'll need to use it there so it **needs** to be reusable. 

If not, even then to me it makes sense to make it reusable just in case...

But feel free to resolve this. Don't want to be dogmatic but makes no sense to me to keep it here..

File: suite-common/wallet-utils/src/confirmAddressOnDevice.ts (hunk 5 above)
Comment (PeKne) [reply suite-common/wallet-utils/src/confirmAddressOnDevice.ts]: thunk created in 9aa7b2e8a4

File: suite-common/wallet-utils/src/confirmAddressOnDevice.ts (hunk 6 above)
Comment (PeKne) [reply suite-common/wallet-utils/src/confirmAddressOnDevice.ts]: Will not be changed after mutual discussion

File: suite-native/receive/src/components/ConfirmOnTrezorImage.tsx (hunk 9 above)
Comment (juriczech) [reply suite-native/receive/src/components/ConfirmOnTrezorImage.tsx:10]: For that you can just remove paddingBottom in Screen.tsx, is that not a sufficient solution? If we have design where we need to avoid the bottom padding, we should avoid it, not hack it this way. If we need to use this in other screen, we would have to do the same thing again.

File: suite-common/wallet-core/src/device/deviceThunks.ts [hunk 23]
```diff
@@ -444,3 +454,63 @@ export const createImportedDeviceThunk = createThunk(
         }
     },
 );
+
+export const confirmAddressOnDeviceThunk = createThunk(
```
Comment (Nodonisko) [inline suite-common/wallet-core/src/device/deviceThunks.ts:459]: Why do we need new method? Isn't this already solved somehow in desktop?

**Review by Nodonisko — CHANGES_REQUESTED** [review]

File: suite-common/wallet-core/src/device/deviceThunks.ts (hunk 23 above)
Comment (PeKne) [reply suite-common/wallet-core/src/device/deviceThunks.ts:459]: I've extracted the desktop logic from `packages/suite` and moved it into `suite-common`. This thunk is basically refactor of the desktop method.

**Review by Nodonisko — APPROVED** [review]

File: suite-common/wallet-utils/src/cardanoUtils.ts [hunk 24]
```diff
@@ -1,5 +1,23 @@
...
+import {
+    amountToSatoshi,
+    formatAmount,
+    formatNetworkAmount,
+    networkAmountToSatoshi,
+} from './accountUtils';
+
 // NOTE: This was moved from packages/suite/src/utils/wallet/cardanoUtils.ts
 // Reason for that is that we need to use it in @suite-common/ package, but we cannot import
 // @fivebinaries/coin-selection there, because it will break mobile app bundling.
```
Comment (juriczech) [inline suite-common/wallet-utils/src/cardanoUtils.ts]: This comment is redundant now, would you please remove it?

**Review by juriczech — APPROVED** [review]

File: suite-common/wallet-utils/src/cardanoUtils.ts (hunk 24 above)
Comment (PeKne) [reply suite-common/wallet-utils/src/cardanoUtils.ts]: Sure [337bbad](https://github.com/trezor/trezor-suite/pull/9844/commits/337bbad74c6303f610548e54802ec369dc246012)

**Review by vytick — APPROVED** [review]
🚢
